pkg/reconciler/trigger/trigger.go

```diff
@@ -301,3 +365,25 @@ func (r *Reconciler) reconcileSubscription(ctx context.Context, t *v1alpha1.Trig
 	}
 	return newSub, nil
 }
+
+func getObjRefFromDependencyAnnotation(dependencyAnnotation string) (corev1.ObjectReference, error) {
+	var objectRef corev1.ObjectReference
+	if err := json.Unmarshal([]byte(dependencyAnnotation), &objectRef); err != nil {
+		return objectRef, err
+	}
+	return objectRef, nil
+}
+
+func validateDependencyAnnotation(depObjRef corev1.ObjectReference) error {
+	var result *multierror.Error
+	if depObjRef.Namespace == "" {
+		result = multierror.Append(result, fmt.Errorf("failed to validate dependency annotation: Namespace should not be empty"))
```

Comment:
Since we only allow dependencies in the same namespace, I think we should validate that one of these is true:
- depObjRef.Namespace is empty
- depObjRef.Namespace is present AND depObjRef.Namespace == trigger namespace